cli/src/main/kotlin/commands/UploadCurationsCommand.kt

```diff
@@ -79,39 +78,26 @@ class UploadCurationsCommand : CliktCommand(
 
     private val service by lazy { ClearlyDefinedService.create(server, OkHttpClientHelper.buildClient()) }
 
-    private fun <T> executeApiCall(call: suspend ClearlyDefinedService.() -> T): Result<T> =
+    private fun <S, T> S.call(block: suspend S.() -> T): T =
```

Comment:
Implementing this as an extension function indeed simplifies the code to call the service and makes it easier to read.

I just ask myself whether the generic `S` parameter only complicates things. As the function is not shared between different Retrofit service implementations and there is only a single service involved, it may be easier to grasp to insert the concrete service type here (`ClearlyDefinedService` in this case). There are still differences between the `call()` functions for different services, so aligning the function signature may not have that big benefit.

Reply:
> As the function is not shared between different Retrofit service implementations

The ultimate goal of course is to do exactly that, but I wasn't able to come up with a nice solution for the `errorTile` handling while looking at this in a time-boxed manner, so I've just stopped at this (intermediate) step. I've clarified that in the commit message.